webapp/packages/plugin-sql-editor-navigation-tab/src/SqlEditorTab.tsx

```diff
@@ -42,13 +44,15 @@ export const SqlEditorTab: TabHandlerTabComponent<ISqlEditorTabState> = observer
   const name = getSqlEditorName(tab.handlerState, dataSource, connection);
   const icon = dataSource?.icon ?? '/icons/sql_script_m.svg';
   const saved = dataSource?.isSaved !== false;
+  const isReadonly = Boolean(dataSource?.isReadonly());
 
   const handleSelect = ({ tabId }: ITabData<any>) => onSelect(tabId);
   const handleClose = onClose ? ({ tabId }: ITabData<any>) => onClose(tabId) : undefined;
 
   return styled(useStyles(style))(
     <Tab tabId={tab.id} style={style} title={name} menuContext={tabMenuContext} onOpen={handleSelect} onClose={handleClose}>
       <TabIcon icon={icon} />
+      {isReadonly && <IconOrImage title={translate('data_grid_table_readonly_tooltip')} icon="/icons/lock.png" className={s(sqlEditorTabStyles, { readonlyIcon: true })} />}
```

Comment:
Please do not use locale tokens from different packages and contexts. Probably we want to create a common ui "ui_readonly" token in core-localization package